cmd/ndm_daemonset/probe/sysfsprobe.go

```diff
@@ -163,4 +172,35 @@ func (cp *sysfsProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevic
 		klog.V(4).Infof("blockdevice path: %s drive type :%s filled by sysfs probe.",
 			blockDevice.DevPath, blockDevice.DeviceAttributes.DriveType)
 	}
+
+	// check if the device is being monitored by for size changes.
+	// add if not being watched
+	if len(cp.watchman.Find(func(f *watchman.File) bool {
+		return f.GetTag() == blockDevice.DevPath
+	})) == 0 {
+		newFile, err := watchman.NewFile(filepath.Join(blockDevice.SysPath, "size"),
+			watchman.WithReader(watchman.ReadFile),
+			watchman.WithTag(blockDevice.DevPath))
+		if err != nil {
+			klog.Error("failed to get size file for size change detection")
+		}
+		if err = cp.watchman.AddFile(newFile); err != nil {
+			klog.Error("failed to watch device size fiel for changes")
+		}
+	}
+}
+
+func (cp *sysfsProbe) listenSizeChanges() {
+	go func() {
+		events, errs := cp.watchman.Start()
+		defer cp.watchman.Stop()
+		for {
+			select {
+			case event := <-events:
+				fmt.Println(event)
+			case err := <-errs:
```

Comment:
Was this introduced so that we get errors though channel instead of logging. Or is there some other purpose for this?